### PR TITLE
PR for eip-message-construction 

### PR DESCRIPTION
**Pull request by SavithriNandadasa**

### CODE REVIEW

File: guide/Ballerina.toml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+[project]
+org-name = "savithri"
+version = "0.0.1"
+
```
Comment (VijithaEkanayake) [inline guide/Ballerina.toml]: Extra newline

File: guide/ballerina-internal.log.1 [hunk 2]
```diff
@@ -0,0 +1,21 @@
+[2018-08-04 16:18:31,711] ERROR {org.ballerinalang.net.http.HttpConnectorPortBindingListener} - Error in http endpoint 
+java.net.BindException: Address already in use
+	at sun.nio.ch.Net.bind0(Native Method)
+	at sun.nio.ch.Net.bind(Net.java:433)
```
Comment (VijithaEkanayake) [inline guide/ballerina-internal.log.1]: we don't need this ballerina-internal.log file, do we?

File: guide/phone_order_delivery_service/phone_order_delivery_service.bal [hunk 3]
```diff
@@ -0,0 +1,230 @@
+import ballerina/log;
+import ballerina/http;
+import ballerina/jms;
+
```
Comment (VijithaEkanayake) [inline guide/phone_order_delivery_service/phone_order_delivery_service.bal]: Extra newline

Comment (SavithriNandadasa) [reply guide/phone_order_delivery_service/phone_order_delivery_service.bal]: Removed all extra new lines

Comment (SavithriNandadasa) [reply guide/phone_order_delivery_service/phone_order_delivery_service.bal]: Rmoved internal error log files.

File: guide/phone_order_delivery_service/phone_order_delivery_service.bal [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+// 'jms:Connection' definition
+
+// 'jms:Session' definition
+
+// 'jms:QueueSender' endpoint definition
+
+//@docker:Config {
+//registry:"ballerina.guides.io",
+//name:"phone_order_delivery_service",
+//tag:"v1.0"
```
Comment (Manuri) [inline guide/phone_order_delivery_service/phone_order_delivery_service.bal]: Have you tested with docker..? Might need to add baseImage due to https://github.com/ballerinax/docker/issues/117

File: guide/phone_order_delivery_service/test/phone_order_delivery_test.bal [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+// KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/http;
+import ballerina/test;
+
+// Function to test 'placeOrder' resource
+@test:Config
+
```
Comment (Manuri) [inline guide/phone_order_delivery_service/test/phone_order_delivery_test.bal]: Do we need a new line here?

File: guide/phone_order_delivery_service/phone_order_delivery_service.bal [hunk 6]
```diff
@@ -0,0 +1,218 @@
...
+//@docker:Expose{}
+//endpoint http:Listener listener {
+//port:9091
+//};
+
+// phone store service, which allows users to order phones online for delivery
+//@http:ServiceConfig {basePath:"/phonestore1"}
+//service<http:Service> phone_order_delivery_service bind listener {
+
+type phoneDeliver record {
```
Comment (hemikak) [inline guide/phone_order_delivery_service/phone_order_delivery_service.bal]: We follow the norm where record names are upper camelcase. Example: https://ballerina.io/learn/by-example/records.html

File: guide/phone_order_delivery_service/phone_order_delivery_service.bal [hunk 7]
```diff
@@ -0,0 +1,218 @@
...
+    port:9091
+};
+
+@http:ServiceConfig {basePath:"/deliveryDetails"}
+
+// phone store service, which allows users to order phones online for delivery
+service<http:Service> phone_order_delivery_service bind deliveryEP {
+    // Resource that allows users to place an order for a phone
+    @http:ResourceConfig { consumes: ["application/json"],
+        produces: ["application/json"] }
```
Comment (hemikak) [inline guide/phone_order_delivery_service/phone_order_delivery_service.bal]: Can we break the annotation attributes to separate lines ?

File: guide/phone_order_delivery_service/phone_order_delivery_service.bal [hunk 8]
```diff
@@ -0,0 +1,218 @@
...
+
```
Comment (hemikak) [inline guide/phone_order_delivery_service/phone_order_delivery_service.bal]: Remove the empty line.

File: guide/phone_store/phone_store_service.bal [hunk 9]
```diff
@@ -0,0 +1,241 @@
...
+jms:Connection conn = new({
+        initialContextFactory: "org.apache.activemq.jndi.ActiveMQInitialContextFactory",
+        providerUrl: "tcp://localhost:61616"
+    });
+    
+// Initialize a JMS session on top of the created connection
+jms:Session jmsSession2 = new(conn, {
+        // Optional property. Defaults to AUTO_ACKNOWLEDGE
+        acknowledgementMode: "AUTO_ACKNOWLEDGE"
+    });
```
Comment (hemikak) [inline guide/phone_store/phone_store_service.bal]: Indentation issue

File: guide/phone_store/phone_store_service.bal [hunk 10]
```diff
@@ -0,0 +1,241 @@
...
+        http:Response response;
+        // Send json array 'phoneInventory' as the response, which contains all the available phones
+        response.setJsonPayload(phoneInventory);
+        _ = client -> respond(response);
+    }
+}
+jms:Connection conn = new({
+        initialContextFactory: "org.apache.activemq.jndi.ActiveMQInitialContextFactory",
+        providerUrl: "tcp://localhost:61616"
+    });
```
Comment (hemikak) [inline guide/phone_store/phone_store_service.bal]: Indentation issue here

File: guide/phone_order_delivery_service/phone_order_delivery_service.bal [hunk 11]
```diff
@@ -0,0 +1,218 @@
...
+        json responseMessage;
+        // If the requested phone is available, then add the order to the 'OrderQueue'
+        if (isPhoneAvailable) {
+            var phoneDeliverDetails = check <json>newDeliver;
+            // Create a JMS message
+
+            jms:Message queueMessage2 = check jmsSession.createTextMessage(phoneDeliverDetails.toString());
+
+            log:printInfo("order Delivery details  added to the delivery  Queue; CustomerName: '" + newDeliver.customerName +
```
Comment (hemikak) [inline guide/phone_order_delivery_service/phone_order_delivery_service.bal]: info logs messages are inconsistent.

Shall we change 'order Delivery details  added to the delivery  Queue' to 'Order delivery details added to the delivery queue' ? Please check other logs as well.

File: guide/phone_store/phone_store_service.bal [hunk 12]
```diff
@@ -0,0 +1,241 @@
...
+// phone store service, which allows users to order phones online for delivery
+@http:ServiceConfig {basePath:"/phonestore"}
+service<http:Service> phone_store_service bind listener {
+    // Resource that allows users to place an order for a phone
+    @http:ResourceConfig { methods: ["POST"], consumes: ["application/json"],
+        produces: ["application/json"] }
+
+    placeOrder(endpoint caller, http:Request request) {
+
+        backendreq= untaint request;
```
Comment (Manuri) [inline guide/phone_store/phone_store_service.bal]: Fix formatting. Need to check other places also for this kind of issues.

File: guide/phone_store/phone_store_service.bal [hunk 13]
```diff
@@ -0,0 +1,241 @@
...
+        log:printInfo("Order Details: " + stringPayload);
+
+        http:Request enrichedreq = backendreq;
+        var clientResponse = phone_order_delivery_serviceEP->forward("/", enrichedreq);
+        match clientResponse {
+            http:Response res => {
+                log:printInfo("Order Details have sent to phone_order_delivery_service.");
+            }
+            error err => {
+                log:printInfo("forward error..................");
```
Comment (Manuri) [inline guide/phone_store/phone_store_service.bal]: Can we add a proper error message with the actual error message combined?

File: guide/phone_store/test/phone_store_service_test.bal [hunk 14]
```diff
@@ -0,0 +1,62 @@
...
+
+@test:Config
+function testResourceGetPhoneList() {
+
+    endpoint http:Client httpEndpoint { url:"http://localhost:9090/phonestore" };
+    // Initialize the empty http request
+    http:Request req;
+
+    // Send a 'get' request and obtain the response
+    http:Response response = check httpEndpoint->get("/getPhoneList", message = req);
```
Comment (daneshk) [inline guide/phone_store/test/phone_store_service_test.bal]: We don't need to initialize empty request here. We can call like,
http:Response response = check httpEndpoint->get("/getPhoneList");
Please check

File: guide/phone_store/phone_store_service.bal [hunk 15]
```diff
@@ -0,0 +1,241 @@
...
+    port:9090
+};
+
+// phone store service, which allows users to order phones online for delivery
+@http:ServiceConfig {basePath:"/phonestore"}
+service<http:Service> phone_store_service bind listener {
+    // Resource that allows users to place an order for a phone
+    @http:ResourceConfig { methods: ["POST"], consumes: ["application/json"],
+        produces: ["application/json"] }
+
```
Comment (daneshk) [inline guide/phone_store/phone_store_service.bal]: Can you remove newline after annotation.

File: guide/phone_order_delivery_service/phone_order_delivery_service.bal [hunk 16]
```diff
@@ -0,0 +1,218 @@
+import ballerina/log;
```
Comment (Anoukh) [inline guide/phone_order_delivery_service/phone_order_delivery_service.bal]: Shall we add the licence header for all files?

File: guide/phone_store/phone_store_service.bal [hunk 17]
```diff
@@ -0,0 +1,241 @@
...
+            // Send the message to the JMS queue
+            _ = jmsProducer -> send(queueMessage);
+
+            // Construct a success message for the response
+            responseMessage = {"Message":"Your order is successfully placed. Ordered phone will be delivered soon"};
+        }
+        else {
+            // If phone is not available, construct a proper response message to notify user
+            responseMessage = {"Message":"Requested phone not available"};
```
Comment (daneshk) [inline guide/phone_store/phone_store_service.bal]: Is it correct to response to caller with 200 OK statuscode here. If we do not specify status code, 200 OK sets as default.

File: guide/phone_store/phone_store_service.bal [hunk 18]
```diff
@@ -0,0 +1,241 @@
...
+};
+
+// Service endpoint
+endpoint http:Listener listener {
+    port:9090
+};
+
+// phone store service, which allows users to order phones online for delivery
+@http:ServiceConfig {basePath:"/phonestore"}
+service<http:Service> phone_store_service bind listener {
```
Comment (hemikak) [inline guide/phone_store/phone_store_service.bal]: phoneStoreService is better than using underscore

File: guide/phone_order_delivery_service/phone_order_delivery_service.bal [hunk 19]
```diff
@@ -0,0 +1,218 @@
...
+
+// Service endpoint
+endpoint http:Listener deliveryEP {
+    port:9091
+};
+
+@http:ServiceConfig {basePath:"/deliveryDetails"}
+
+// phone store service, which allows users to order phones online for delivery
+service<http:Service> phone_order_delivery_service bind deliveryEP {
```
Comment (MaryamZi) [inline guide/phone_order_delivery_service/phone_order_delivery_service.bal]: We don't usually use the `_` with Ballerina identifiers. Shall we use camel case?

File: guide/phone_order_delivery_service/phone_order_delivery_service.bal [hunk 20]
```diff
@@ -0,0 +1,218 @@
...
+    // Resource that allows users to place an order for a phone
+    @http:ResourceConfig { consumes: ["application/json"],
+        produces: ["application/json"] }
+
+    sendDelivery(endpoint caller, http:Request enrichedreq) {
+        http:Response response;
+        phoneDeliver newDeliver;
+        json reqPayload;
+
+        log:printInfo(" Order Details have received from phone_store_service");
```
Comment (MaryamZi) [inline guide/phone_order_delivery_service/phone_order_delivery_service.bal]: extra space at the beginning, and should this be "Order details have **been** received from the phone store service" or "Order details received from the phone store service"?

File: guide/phone_order_delivery_service/phone_order_delivery_service.bal [hunk 21]
```diff
@@ -0,0 +1,218 @@
...
+
+        // Try parsing the JSON payload from the request
+        match  enrichedreq.getJsonPayload() {
+            // Valid JSON payload
+            json payload => reqPayload = payload;
+            // NOT a valid JSON payload
+            any => {
+                response.statusCode = 400;
+                response.setJsonPayload({"Message":"Invalid payload - Not a valid JSON payload"});
+                _ = caller -> respond(response);
```
Comment (MaryamZi) [inline guide/phone_order_delivery_service/phone_order_delivery_service.bal]: Shall we handle the possible error here, similar to https://github.com/ballerina-platform/ballerina-examples/blob/master/examples/hello-world-service/hello_world_service.bal#L14 ? 

Also we don't use spaces around `->` when it's an endpoint action. Shall we change to `caller->respond(response)`?

File: guide/phone_order_delivery_service/phone_order_delivery_service.bal [hunk 22]
```diff
@@ -0,0 +1,218 @@
...
+
+service<jms:Consumer> deliverySystem bind jmsConsumer2 {
+    // Triggered whenever an order is added to the 'OrderQueue'
+    onMessage(endpoint consumer, jms:Message message2) {
+        log:printInfo("New order successfilly received from the Delivery Queue");
+        // Retrieve the string payload using native function
+        string stringPayload2 = check message2.getTextMessageContent();
+        log:printInfo("Delivery Details: " + stringPayload2);
+
+        log:printInfo(" Delivery Details sent to the customer successfully");
```
Comment (MaryamZi) [inline guide/phone_order_delivery_service/phone_order_delivery_service.bal]: do we need all these logs here? can't we combine maybe 214 and 216?

Also, typo in line 211, "successf**i**lly"

File: guide/phone_store/phone_store_service.bal (hunk 13 above)
Comment (MaryamZi) [reply guide/phone_store/phone_store_service.bal]: also, shouldn't this be `log:printError`?

File: guide/phone_store/phone_store_service.bal [hunk 23]
```diff
@@ -0,0 +1,241 @@
...
+            http:Response res => {
+                log:printInfo("Order Details have sent to phone_order_delivery_service.");
+            }
+            error err => {
+                log:printInfo("forward error..................");
+            }
+        }
+    }
+}
+endpoint http:Client phone_order_delivery_serviceEP {
```
Comment (pubudu91) [inline guide/phone_store/phone_store_service.bal]: Snake case is not really used for variable or endpoint names.

Comment (pubudu91) [reply guide/phone_store/phone_store_service.bal]: Also, better if this client endpoint is defined before its use. i.e., move this EP up.

File: README.md [hunk 24]
```diff
@@ -1 +1,966 @@
...
+
+```bash
+   curl -v -X POST -d \
+   '{"Name":"John", "Address":"20, Palm Grove, Colombo, Sri Lanka", 
+   "ContactNumber":"+94718930874", "PhoneName":"Apple:190000"}' \
+   "http://localhost:9090/phonestore/placeOrder" -H "Content-Type:application/json"
+   
+```
+
+  The bookstoreService sends a response similar to the following.
```
Comment (pubudu91) [inline README.md]: Back ticks for formatting

File: README.md [hunk 25]
```diff
@@ -1 +1,966 @@
...
+```
+   < HTTP/1.1 200 OK
+   {"Message":"Your order is successfully placed. Ordered book will be delivered soon"}
```
Comment (pubudu91) [inline README.md]: is -> was

File: README.md [hunk 26]
```diff
@@ -1 +1,966 @@
...
+```
+
+  Sample Log Messages:
+```bash
+
+  INFO  [phone_store_service] - order will be added to the order  Queue; CustomerName: 'Bob', OrderedPhone: 'Apple:190000'; 
+  INFO  [phone_store_service] - New order successfilly received from the Order Queue 
+  INFO  [phone_store_service] - Order Details: {"customerName":"John","address":"20, Palm Grove, Colombo, Sri Lanka","contactNumber":"+94718930874","orderedPhoneName":"Apple:190000"} 
+  
+  Order Details have sent to phone_order_delivery_service.
```
Comment (pubudu91) [inline README.md]: Order details were sent to `phone_order_delivery_service`.

File: README.md [hunk 27]
```diff
@@ -1 +1,966 @@
...
+
+
+  Order Details have received from phone_store_service
```
Comment (pubudu91) [inline README.md]: Received order details from the `phone_store_service`.

Comment (pubudu91) [reply README.md]: Is there a reason for capitalizing order details as Order Details?

File: README.md (hunk 26 above)
Comment (pubudu91) [reply README.md]: If this is a log message, please correct in other relevant places as well

File: README.md [hunk 28]
```diff
@@ -1 +1,966 @@
...
+  
+  INFO  [phone_order_delivery_service] - order Delivery details  added to the delivery  Queue; CustomerName: 'Bob', OrderedPhone: 'Apple:190000';
```
Comment (pubudu91) [inline README.md]: Delivery -> delivery

Need to use cases consistently across the docs. Check in other places as well. Also, I think the 'Q' in Queue can be simple as well right?

File: README.md [hunk 29]
```diff
@@ -1 +1,966 @@
...
+
+In Ballerina, the unit test cases should be in the same package inside a folder named `tests`.  When writing the test functions the below convention should be followed.
+- Test functions should be annotated with `@test:Config`. See the below example.
+
+```ballerina
+   @test:Config
+   function testResourcePlaceOrder() {
+```
+  
+This guide contains unit test cases for each resource available in the 'bookstore_service' implemented above.
```
Comment (pubudu91) [inline README.md]: Should be back ticks, not quotes.

File: README.md [hunk 30]
```diff
@@ -1 +1,966 @@
...
+   ballerina: initiating service(s) in 'phone_store.balx' 
+   ballerina: started HTTP/WS endpoint 0.0.0.0:9090
+   
+   ballerina: initiating service(s) in 'phone_order_delivery_service.balx' 
+   ballerina: started HTTP/WS endpoint 0.0.0.0:9091
+```
+### Deploying on Docker
+
+You can run the service that we developed above as a Docker container.
+As Ballerina platform includes [Ballerina_Docker_Extension](https://github.com/ballerinax/docker), which offers native support for running ballerina programs on containers,
```
Comment (pubudu91) [inline README.md]: ballerina -> Ballerina

Correct in all the places

File: guide/phone_order_delivery_service/phone_order_delivery_service.bal [hunk 31]
```diff
@@ -0,0 +1,218 @@
...
+//service<http:Service> phone_order_delivery_service bind listener {
+
+type phoneDeliver record {
+    string customerName;
+    string address;
+    string contactNumber;
+    string deliveryPhoneName;
+};
+
+json[] phoneInventory = ["Apple:190000", "Samsung:150000", "Nokia:80000", "HTC:40000", "Huawei:100000"];
```
[the remark this answers is not in the document]
Comment (pubudu91) [reply guide/phone_order_delivery_service/phone_order_delivery_service.bal]: Shouldn't this be an array like the following for example:
```ballerina
json[] phoneInventory = [{model: "Apple", price: 190000}, {model: "Samsung", price:150000}, {model:"Nokia", price:80000}, {model:"HTC", price:40000}, {model:"Huawei", price:100000}];
```

File: README.md [hunk 32]
```diff
@@ -1 +1,966 @@
-# eip-message-construction
\ No newline at end of file
+[![Build Status](https://travis-ci.org/SavithriNandadasa/message_construction_patterns.svg?branch=master)](https://travis-ci.org/SavithriNandadasa/message_construction_patterns)
+
+# Message Construction Patterns
+
+Message construction patterns describes the creation of message content that travel across the messaging system.Java Message
```
[the remark this answers is not in the document]
Comment (MaryamZi) [reply README.md]: +1 for "a messaging system" or "messaging systems"

not sure if "travel across" is the appropriate term here though? 

Also, the second sentence (regarding JMS) seems a bit disconnected from the first?

File: README.md [hunk 33]
```diff
@@ -1 +1,966 @@
...
+- Once you successfully build the Docker image, you can run it with the `` docker run`` command that is shown in the previous step.  
+
+```bash
+  docker run -d -p 9090:9090 ballerina.guides.io/phone_store_service:v1.0
+  docker run -d -p 9091:9091 ballerina.guides.io/phone_order_delivery_service:v1.0  
+```
+
+   Here we run the Docker image with flag`` -p <host_port>:<container_port>`` so that we use the host port 9090 and the container port 9090. Therefore you can access the service through the host port. 
+
+- Verify docker container is running with the use of `` $ docker ps``. The status of the Docker container should be shown as 'Up'.
```
Comment (pubudu91) [inline README.md]: docker -> Docker

File: README.md [hunk 34]
```diff
@@ -1 +1,966 @@
...
+
+- Since this guide requires `ActiveMQ` as a prerequisite, you need an additional step to create a pod for `ActiveMQ` and use it with our sample.  
+
+- Navigate to `message_construction_patterns/resources` directory and run the below command to create the ActiveMQ pod by creating a deployment and service for ActiveMQ. You can find the deployment descriptor and service descriptor in the `./resources/kubernetes` folder.
+
+```bash
+   $ kubectl create -f ./kubernetes/
+```
+
+- Now let's see how we can deploy the `phone_store_service` on Kubernetes. We need to import `` ballerinax/kubernetes; `` and use ``@kubernetes``annotations as shown below to enable kubernetes deployment.
```
Comment (pubudu91) [inline README.md]: Remove the semi colon. Also, no need for double back ticks right?

Leave a space between "@kubernetes" and "annotations".

kubernetes -> Kubernetes. Check and fix in all relevant places

File: README.md [hunk 35]
```diff
@@ -1 +1,966 @@
...
+- Verify docker container is running with the use of `` $ docker ps``. The status of the Docker container should be shown as 'Up'. 
+
+- You can access the service using the same curl commands that we've used above.
+```bash
+    curl -v -X POST -d \
+   '{"Name":"John", "Address":"20, Palm Grove, Colombo, Sri Lanka", 
+   "ContactNumber":"+94718930874", "PhoneName":"Apple:190000"}' \
+   "http://localhost:9090/phonestore/placeOrder" -H "Content-Type:application/json"
+```
+### Deploying on Kubernetes
```
Comment (pubudu91) [inline README.md]: The instructions for Linux users to use minikube is missing in this section. See https://github.com/ballerina-guides/restful-service/blob/master/README.md#deploying-on-kubernetes

File: README.md [hunk 36]
```diff
@@ -1 +1,966 @@
...
+   Run following command to deploy kubernetes artifacts:  
+   kubectl apply -f ./target/phone_store_service/kubernetes
+   
+   @kubernetes:Service                      - complete 1/1
+   @kubernetes:Ingress                      - complete 1/1
+   @kubernetes:Docker                       - complete 3/3 
+   @kubernetes:Deployment                   - complete 1/1
+  
+   Run following command to deploy kubernetes artifacts:  
+   kubectl apply -f ./target/phone_order_delivery_service/kubernetes
```
Comment (pubudu91) [inline README.md]: IIRC, the build output has indented levels? Shall we include the exact build output as it appears on your terminal?

Comment (pubudu91) [reply README.md]: Check and fix in all relevant places

File: README.md [hunk 37]
```diff
@@ -1 +1,966 @@
...
+  
+   
+```
+
+- You can verify that the Docker image that we specified in `` @kubernetes:Deployment `` is created, by using `` docker images ``.
```
Comment (pubudu91) [inline README.md]: rephrase to "Use the `docker images` command to verify whether the Docker image that you specified in `@kubernetes:Deployment` was created."

File: README.md [hunk 38]
```diff
@@ -1 +1,966 @@
...
+   @kubernetes:Deployment                   - complete 1/1
+  
+   Run following command to deploy kubernetes artifacts:  
+   kubectl apply -f ./target/phone_order_delivery_service/kubernetes
+  
+   
+```
+
+- You can verify that the Docker image that we specified in `` @kubernetes:Deployment `` is created, by using `` docker images ``. 
+- Also the Kubernetes artifacts related our service, will be generated under `` ./target/phone_store_service/kubernetes`` and ``/target/phone_order_delivery_service/kubernetes``.
```
Comment (pubudu91) [inline README.md]: Shall we rephrase this to "The Kubernetes artifacts related to the service are generated in `./target/phone_store_service/kubernetes` and `./target/phone_order_delivery_service/kubernetes` directories."

File: README.md [hunk 39]
```diff
@@ -1 +1,966 @@
...
+- You can verify Kubernetes deployment, service and ingress are running properly, by using following Kubernetes commands. 
+
+```bash
+   $ kubectl get service
+   $ kubectl get deploy
+   $ kubectl get pods
+   $ kubectl get ingress
+```
+
+- If everything is successfully deployed, you can invoke the service either via Node port or ingress.
```
Comment (pubudu91) [inline README.md]: Shall we rephrase this to "If all artifacts are successfully deployed, you can invoke the service either via Node Port or Ingress."? Refer to https://github.com/ballerina-guides/restful-service/blob/master/README.md#deploying-on-kubernetes as this was reviewed and refined by the doc team.

File: README.md [hunk 40]
```diff
@@ -1 +1,966 @@
...
+```
+
+Access the service 
+```bash
+    curl -v -X POST -d \
+   '{"Name":"John", "Address":"20, Palm Grove, Colombo, Sri Lanka", 
+   "ContactNumber":"+94718930874", "PhoneName":"Apple:190000"}' \
+   "http://localhost:9090/phonestore/placeOrder" -H "Content-Type:application/json" 
+```
+## Observability
```
Comment (pubudu91) [inline README.md]: Please refer to https://github.com/ballerina-guides/restful-service/blob/master/README.md#observability and refine this section accordingly.

File: README.md [hunk 41]
```diff
@@ -1 +1,960 @@
...
+message_construction_patterns
+ └── guide
+      ├── phone_store_service
+      │    ├── phone_store_service.bal
+      │    └── tests
+      │         └── phone_store_service.bal
+      └── phone_order_delivery_service
+           ├──order_delivery_service.bal
+           └── tests
+                └── phone_order_delivery_service_test.bal
```
Comment (Anoukh) [inline README.md]: Can you recheck if this directory structure is correct?

File: guide/phone_order_delivery_service/phone_order_delivery_service.bal (hunk 4 above)
Comment (SavithriNandadasa) [reply guide/phone_order_delivery_service/phone_order_delivery_service.bal]: added baseimage

File: guide/phone_order_delivery_service/test/phone_order_delivery_test.bal (hunk 5 above)
Comment (SavithriNandadasa) [reply guide/phone_order_delivery_service/test/phone_order_delivery_test.bal]: removed new line

File: guide/phone_order_delivery_service/phone_order_delivery_service.bal (hunk 6 above)
Comment (SavithriNandadasa) [reply guide/phone_order_delivery_service/phone_order_delivery_service.bal]: change the record name

File: guide/phone_order_delivery_service/phone_order_delivery_service.bal (hunk 7 above)
Comment (SavithriNandadasa) [reply guide/phone_order_delivery_service/phone_order_delivery_service.bal]: done

File: guide/phone_order_delivery_service/phone_order_delivery_service.bal (hunk 8 above)
Comment (SavithriNandadasa) [reply guide/phone_order_delivery_service/phone_order_delivery_service.bal]: done

File: guide/phone_store/phone_store_service.bal (hunk 9 above)
Comment (SavithriNandadasa) [reply guide/phone_store/phone_store_service.bal]: done

File: guide/phone_store/phone_store_service.bal (hunk 10 above)
Comment (SavithriNandadasa) [reply guide/phone_store/phone_store_service.bal]: done

File: guide/phone_order_delivery_service/phone_order_delivery_service.bal (hunk 11 above)
Comment (SavithriNandadasa) [reply guide/phone_order_delivery_service/phone_order_delivery_service.bal]: done

File: guide/phone_order_delivery_service/phone_order_delivery_service.bal [hunk 42]
```diff
@@ -0,0 +1,218 @@
+import ballerina/log;
+import ballerina/http;
+import ballerina/jms;
```
[the remark this answers is not in the document]
Comment (SavithriNandadasa) [reply guide/phone_order_delivery_service/phone_order_delivery_service.bal]: done

File: guide/phone_store/phone_store_service.bal (hunk 12 above)
Comment (SavithriNandadasa) [reply guide/phone_store/phone_store_service.bal]: done

File: guide/phone_store/phone_store_service.bal (hunk 13 above)
Comment (SavithriNandadasa) [reply guide/phone_store/phone_store_service.bal]: done

File: guide/phone_order_delivery_service/phone_order_delivery_service.bal (hunk 16 above)
Comment (SavithriNandadasa) [reply guide/phone_order_delivery_service/phone_order_delivery_service.bal]: done

File: guide/phone_store/phone_store_service.bal (hunk 17 above)
Comment (SavithriNandadasa) [reply guide/phone_store/phone_store_service.bal]: done

File: guide/phone_store/phone_store_service.bal (hunk 18 above)
Comment (SavithriNandadasa) [reply guide/phone_store/phone_store_service.bal]: done

File: guide/phone_order_delivery_service/phone_order_delivery_service.bal (hunk 19 above)
Comment (SavithriNandadasa) [reply guide/phone_order_delivery_service/phone_order_delivery_service.bal]: done

File: guide/phone_order_delivery_service/phone_order_delivery_service.bal (hunk 20 above)
Comment (SavithriNandadasa) [reply guide/phone_order_delivery_service/phone_order_delivery_service.bal]: done

File: guide/phone_order_delivery_service/phone_order_delivery_service.bal (hunk 21 above)
Comment (SavithriNandadasa) [reply guide/phone_order_delivery_service/phone_order_delivery_service.bal]: done

File: guide/phone_order_delivery_service/phone_order_delivery_service.bal (hunk 22 above)
Comment (SavithriNandadasa) [reply guide/phone_order_delivery_service/phone_order_delivery_service.bal]: done

File: guide/phone_order_delivery_service/phone_order_delivery_service.bal (hunk 31 above)
[the remark this answers is not in the document]
Comment (SavithriNandadasa) [reply guide/phone_order_delivery_service/phone_order_delivery_service.bal]: done

File: README.md [hunk 43]
```diff
@@ -1 +1,966 @@
-# eip-message-construction
\ No newline at end of file
+[![Build Status](https://travis-ci.org/SavithriNandadasa/message_construction_patterns.svg?branch=master)](https://travis-ci.org/SavithriNandadasa/message_construction_patterns)
```
[the remark this answers is not in the document]
Comment (SavithriNandadasa) [reply README.md]: done

File: README.md (hunk 40 above)
Comment (SavithriNandadasa) [reply README.md]: done
